frontend/src/app/cards/cards.component.ts

```
@@ -80,16 +81,27 @@ export class CardsComponent implements OnInit {
             if (entry !== undefined) {
                 this.selectedEntry = entry;
                 this.cardOverlayIsShown = true;
-                const selectedElement = document.getElementById('card-' + entry.id);
-                const selectedElementY = selectedElement.offsetTop;
-                scroll(0, selectedElementY + -250);
             } else {
                 this.selectedEntry = undefined;
                 this.cardOverlayIsShown = false;
             }
         });
     }
 
+    ngAfterViewChecked(): void {
+        if (this.selectedEntry) {
+            const selectedElement = document.getElementById('card-' + this.selectedEntry.id);
+            if (selectedElement) {
+                const selectedElementY = selectedElement.offsetTop;
```

Hier könnte man die Verschachtelung etwas reduzieren, wenn man die ifs umdreht, also z.B.
if (!this.selectedEntry) { return; }

Damit wird auch leichter verständlich, was hier gemeint ist

---

Versteh ich jetzt nicht ganz

---

Wie schon weiter unten erklärt, geht es darum, die Verschachtelungstiefe zu reduzieren, wodurch der code besser verständlich und wartbarer wird.
Im Endeffekt machst du statt
`if (xy) {
if (ab) {
irgendeine logik
}
}`
das hier:
`if (!xy) {
return
}
if (!ab) {
return
}
irgendeine Logik`

Dadurch sieht man besser, was die Anforderungen sind und man hat einen klareren Abbruch, wenn die Felder nicht gesetzt sind. Das Ganze kann man auch unter "fail fast" zusammenfassen.

Nochmal anders erklärt: beim aktuellen Code muss ich die ganze Methode lesen, um rauszufinden, was passiert, wenn this.selectedEntry nicht gesetzt ist. Mit dem umgedrehten if muss ich nur zwei Zeilen lesen, um das zu verstehen. Umgedrehtes if übrigens deshalb, weil die Funktion im IntelliJ, die das macht, "Invert if" heißt ;-)

---

Wie schon weiter unten erklärt, geht es darum, die Verschachtelungstiefe zu reduzieren, wodurch der code besser verständlich und wartbarer wird.
Im Endeffekt machst du statt
`if (xy) {
if (ab) {
irgendeine logik
}
}`
das hier:
`if (!xy) {
return
}
if (!ab) {
return
}
irgendeine Logik`

Dadurch sieht man besser, was die Anforderungen sind und man hat einen klareren Abbruch, wenn die Felder nicht gesetzt sind. Das Ganze kann man auch unter "fail fast" zusammenfassen.

Nochmal anders erklärt: beim aktuellen Code muss ich die ganze Methode lesen, um rauszufinden, was passiert, wenn this.selectedEntry nicht gesetzt ist. Mit dem umgedrehten if muss ich nur zwei Zeilen lesen, um das zu verstehen. Umgedrehtes if übrigens deshalb, weil die Funktion im IntelliJ, die das macht, "Invert if" heißt ;-)

---

Achsooo das hab ich so jetzt auch noch nie gesehen. Danke für den Tipp!!

---

Done